### ISSUE TITLE
takeoff can not work

### ISSUE THREAD

**Issue by xiaobai-cai-hold** [DavidLindtner/px4_ros2_sim#1]
hello, David.
I saw your code in drone.hpp file, I find you write a function that set the uav flight mode.
this->publish_vehicle_command(VehicleCommand::VEHICLE_CMD_NAV_TAKEOFF, 5.5, 1.5); the line you write, but you don't use it
I test it , it doesn't work

I used code to control the multicopte to takeoff  via publish_vehicle_command like you , but I failed, the QGC show use minmum height to takeoff, I guesss I need to set the takeoff_height in my code.

do you have time.I am glad to take with you,thanks.

**Comment by DavidLindtner** [comment]
Thanks for response.
I am currently using offboard flight mode for take off but I am searching how to fix this problem.

**Comment by xiaobai-cai-hold** [comment]
Thanks for your replying, I will share with you when I find soultion to this problem,best wish for you.

**Comment by DavidLindtner** [comment]
I find a solution for setting a drone to takeoff flight mode.
You can use:
this->publish_vehicle_command(VehicleCommand::VEHICLE_CMD_DO_SET_MODE, 1, 4, 2);
You can find it in another repository: https://github.com/DavidLindtner/px4_ros2_missions

**Comment by xiaobai-cai-hold** [comment]
> I find a solution for setting a drone to takeoff flight mode. You can use: this->publish_vehicle_command(VehicleCommand::VEHICLE_CMD_DO_SET_MODE, 1, 4, 2); You can find it in another repository: https://github.com/DavidLindtner/px4_ros2_missions

Thanks for your responsing , this command can set uav takeoff mode, do you test it in simulation? can the uav takeoff successfully?

**Comment by DavidLindtner** [comment]
Yes, it is working in simulation. You need to set take off flight mode and then arm the vehicle. Drone will take off to the minimum takeoff altitude 2.5 m.
